### PR TITLE
chore: add logic to handle serialize image pulls for custom kubelet config

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
This PR adds logic to update AKS Kubelet Config datamodel to include SerializeImagePulls, taken from: https://github.com/kubernetes/kubernetes/blob/5fdab96c6cd7a3f9c179725476c6933757aea2ba/staging/src/k8s.io/kubelet/config/v1beta1/types.go#L489C2-L495C66

This will ensure that when custom kubelet config is specified for k8s 1.31 and above, serialize-image-pulls value is also set in the config file along with other parameters.

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

**Review by Devinwong — APPROVED** [review]

File: pkg/agent/utils.go [hunk 1]
```diff
@@ -465,6 +465,13 @@ func getAKSKubeletConfiguration(kc map[string]string) *datamodel.AKSKubeletConfi
 		ResolverConfig:                 kc["--resolv-conf"],
 		ContainerLogMaxSize:            kc["--container-log-max-size"],
 	}
+
+	// Serialize Image Pulls will only be set for k8s >= 1.31, currently RP doesnt pass this flag
+	// It will starting with k8s 1.31
+	if value, exists := kc["--serialize-image-pulls"]; exists {
+		kubeletConfig.SerializeImagePulls = strToBoolPtr(value)
```
Comment (cameronmeissner) [inline pkg/agent/utils.go:472]: did we also wanna add a UT for this?

File: pkg/agent/datamodel/types.go [hunk 2]
```diff
@@ -2088,6 +2088,13 @@ type AKSKubeletConfiguration struct {
 	Default: []
 	+optional. */
 	AllowedUnsafeSysctls []string `json:"allowedUnsafeSysctls,omitempty"`
+	// serializeImagePulls when enabled, tells the Kubelet to pull images one
```
Comment (djsly) [inline pkg/agent/datamodel/types.go:2091]: by default parallel pull is used ?

**Review by djsly — APPROVED** [review]

File: pkg/agent/datamodel/types.go (hunk 2 above)
Comment (anujmaheshwari1) [reply pkg/agent/datamodel/types.go:2091]: by default, serial pull is used.

File: pkg/agent/utils.go (hunk 1 above)
Comment (anujmaheshwari1) [reply pkg/agent/utils.go:461]: with the original change I had, this was already UTed, but since I reverted it, added UTs specifically for this now

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**